webapp/src/Controller/API/JudgehostController.php

```diff
@@ -1001,11 +1001,13 @@ private function addSingleJudgingRun(
                     break;
                 }
             }
+            $sendJudgingEvent = false;
             if (!$hasNullResults || $lazyEval != DOMJudgeService::EVAL_FULL) {
                 // NOTE: setting endtime here determines in testcases_GET
                 // whether a next testcase will be handed out.
                 $judging->setEndtime(Utils::now());
                 $this->maybeUpdateActiveJudging($judging);
+                $sendJudgingEvent = true;
```

Comment:
We would still send the message when you run `judge remaining`, if I understand correctly?

Reply:
We send the event when the end time changes, which I think is indeed if you judge remain. But that’s fine